app/models/conversion_host.rb

```diff
@@ -68,6 +68,10 @@ def verify_credentials(auth_type = nil, options = {})
         raise MiqException::MiqInvalidCredentialsError, _("Unknown auth type: #{auth.authtype}")
       end
 
+      # Don't connect again if the authentication was valid within the last 15
+      # minutes. This helps reduce unnecessary ssh connections.
+      return true if auth.last_valid_on && auth.last_valid_on > 15.minutes.ago
```

Comment:
Eh, I thought we should validate the auth type before checking if it's valid.

You could argue that this would be a nice method to have baked into either the Authentication model or the mixin, but that's a separate PR.

It only uses `check_ssh_connection` if there's no associated authentication.

Reply:
I think the use of `authentication_status_ok?` essentially resolves your concerns here.